python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -335,17 +344,25 @@ pyAltLaunchKernelBase(const std::string &name, MlirModule module,
   kernelReg();
 
   if (launch) {
-    auto &platform = cudaq::get_platform();
     auto uReturnOffset = static_cast<std::uint64_t>(returnOffset);
-    if (platform.is_remote() || platform.is_emulated()) {
+    if (platform.get_remote_capabilities().isRemoteSimulator) {
+      // Remote simulator - use altLaunchKernel to support returning values.
       auto *wrapper = new cudaq::ArgWrapper{mod, names, rawArgs};
       auto dynamicResult = cudaq::altLaunchKernel(
           name.c_str(), thunk, reinterpret_cast<void *>(wrapper), size,
           uReturnOffset);
       if (dynamicResult.data_buffer || dynamicResult.size)
         throw std::runtime_error("not implemented: support dynamic results");
       delete wrapper;
+    } else if (platform.is_remote() || platform.is_emulated()) {
+      // Quantum devices or their emulation - we can use streamlinedLaunchKernel
```

Comment:
Does `is_remote` mean "quantum hardware device"? That's not very intuitive, if so.

Reply:
Yeah this is very non-intuitive, `is_remote() || is_emulated()` is true for remote simulators, quantum devices, or their emulation. `platform.get_remote_capabilities().isRemoteSimulator` is only true for the remote simulator.

I will add some clarification.